packages/cli/src/controllers/dynamicNodeParameters.controller.ts

```diff
@@ -82,16 +82,24 @@ export class DynamicNodeParametersController {
 		const { path, methodName, filter, paginationToken } = req.query;
 		const { credentials, currentNodeParameters, nodeTypeAndVersion } = req.params;
 		const additionalData = await getBase(req.user.id, currentNodeParameters);
-		return await this.service.getResourceLocatorResults(
-			methodName,
-			path,
-			additionalData,
-			nodeTypeAndVersion,
-			currentNodeParameters,
-			credentials,
-			filter,
-			paginationToken,
-		);
+		try {
+			return await this.service.getResourceLocatorResults(
+				methodName,
+				path,
+				additionalData,
+				nodeTypeAndVersion,
+				currentNodeParameters,
+				credentials,
+				filter,
+				paginationToken,
+			);
+		} catch (error) {
+			if (error instanceof NodeOperationError) {
+				throw new BadRequestError(error.message);
+			}
```

Comment:
Can this throw a `NodeOperationError` unrelated to invalid credentials? How do we know this mapping is always true?

Reply:
Tbh I don't know what the conventions are either - @michael-radency what would you suggest?

Reply:
for nodes if error comes from external api request it's in most cases would be `NodeApiError`, if error happened during operation, like invalid json parse, it would be `NodeOperationError`

so in case of invalid credentials it probably should be `NodeApiError`, but there are exceptions

not sure why we need to add a new error class as `BadRequestError` is a specialized wrapper to throw error with status code 400

Reply:
So would it make sense then to change the [`NodeExecuteFunctions.getCredentials`](https://github.com/n8n-io/n8n/blob/610ead9a3851eeee246313669d0ed9049c736a1a/packages/core/src/NodeExecuteFunctions.ts#L1969-L2101) to throw `NodeApiError` instead? Even tho the error is not coming from an external API. I wouldn't change it to `BadRequestError` as that wouldn't make sense conceptually in a function like `getCredentials`.

Reply:
maybe better to add `errorCode` optional parameter to `NodeOperationErrorOptions` that would set error code on `NodeOperationError`